tutorials/5-docker-intro/index.md

```diff
@@ -0,0 +1,288 @@
+```
+
+It should return:
+
+```txt
+Use "--help category" to get an overview of all categories.
+For all options use the manual or "--help all".
+```
+
+So yes, you now have `curl` and Node on the same container.
+
+### Hello World
+
+What if you wanted to use Node to print `Hello World`? Create the Javascript file that can do that:
+
+```sh
+$ echo "console.log(\"Hello World\");" > test.js
+```
+
+Now pass it to your container:
+
+```sh
+$ docker run -i node:lts-slim test.js
+```
+
+This does not work:
+
+```txt
+Error: Cannot find module '/test.js'
+```
+
+That is because Docker took the words `test.js` and passed them to Node as a string within the context of the container, which has no `test.js`. So the container in effect ran `node test.js`. The file is currently only in your host computer.
+
+Try again:
+
+```sh
+$ docker run -it node:lts-slim < test.js
+```
+
+Here the content of the file is being passed via StdIn. It should still complain:
+
+```txt
+the input device is not a TTY
+```
+
+So remove `-t`:
+
+```sh
+$ docker run -i node:lts-slim < test.js
+```
+
+This time it prints:
+
+```txt
+Hello World
+```
+
+### Sharing folders
+
+Shared folders are called **volumes**. Instead of sending the content of `test.js` to the container via StdIn, it could be more judicious to let the container have access to your file directly. Try again, this time by sharing your local folder (`pwd`) with the container, and mounting it at `/root/temp` inside the container:
+
+```sh
+$ docker run -it -v $(pwd):/root/temp node:lts-slim test.js
+```
+
+It should complain again:
+
+```txt
+Error: Cannot find module '/test.js'
+```
+
+For that, you have to also tell it to work (`-w`) in the right folder: `/root/temp` that you created to access your local files:
+
+```sh
+$ docker run -it -v $(pwd):/root/temp -w /root/temp node:lts-slim test.js
+```
+
+This time it returns:
+
+```txt
+Hello World
+```
+
+## Clean up
+
+You ran quite a few commands. Where did you containers go?
+
+```sh
+$ docker ps --all
+```
+
+You should see a long list like:
+
+```txt
+CONTAINER ID   IMAGE           COMMAND                  CREATED          STATUS                        PORTS     NAMES
+b5930aef7e7a   node:lts-slim   "docker-entrypoint.s…"   2 minutes ago    Exited (0) 2 minutes ago                angry_babbage
+e3f58f2cb118   node:lts-slim   "docker-entrypoint.s…"   5 minutes ago    Exited (0) 5 minutes ago                affectionate_elgamal
+...
+```
+
+All stopped and exited. When you create a container, it is not removed by default so it can be reused (see `docker exec --help`). For now, you will clean up these pointless containers (if you have containers that are not part of this introduction, do not run the command):
+
+```sh
+$ docker container prune
+```
+
+Having a ton of stopped containers is not ideal, which is why when you want to run a container for a single command, the practice is to add `--rm`, like so:
```

Comment:
The practice is to add `--rm` ... because this automatically **removes** the container? Y/"N? And if so, when is the container removed?

Reply:
Yes it removes.
The container is removed at the moment it is stopped and exited.